### PR TITLE
Create Track Game Command

### PR DESCRIPTION
**Pull request by duncanspumpkin**
Should work. Still to test with ai.

### CODE REVIEW

File: src/OpenLoco/src/GameCommands/Track/CreateTrack.h [hunk 1]
```diff
@@ -14,7 +14,8 @@ namespace OpenLoco::GameCommands
             , rotation(regs.bh & 0x3)
             , trackId(regs.dh & 0x3F)
             , mods(regs.di >> 16)
-            , bridge(regs.edx >> 24)
+            , unkFlags((regs.edx >> 20) & 0xF)
```
Comment (AaronVanGeffen) [inline src/OpenLoco/src/GameCommands/Track/CreateTrack.h:17]: Any idea what the consequences could have been of these flags having been missing until now? I suppose their purpose is still unknown, given you've labelled them 'unkFlags'...

Comment (duncanspumpkin) [reply src/OpenLoco/src/GameCommands/Track/CreateTrack.h:17]: They are only used for ai. Not sure exactly what

**Review by AaronVanGeffen — APPROVED** [review]
Constructing track appears to be working fine, including the various kinds of bridges.

Haven't tested AI, so please confirm replay tests pass before merging.

**Comment by duncanspumpkin** [comment]
> Constructing track appears to be working fine, including the various kinds of bridges.
> 
> Haven't tested AI, so please confirm replay tests pass before merging.

I'm getting a desync. There are lots of globals in this code so could be related to that

File: src/OpenLoco/src/GameCommands/Track/CreateTrack.h [hunk 2]
```diff
@@ -13,7 +13,7 @@ namespace OpenLoco::GameCommands
             : pos(regs.ax, regs.cx, regs.di)
             , rotation(regs.bh & 0x3)
             , trackId(regs.dh & 0x3F)
-            , mods(regs.di >> 16)
+            , mods((regs.edi >> 16) & 0xFF)
```
Comment (AaronVanGeffen) [inline src/OpenLoco/src/GameCommands/Track/CreateTrack.h:16]: Incredible. Wow.

**Comment by LeftofZen** [comment]
This is a big one! Things I play-tested and found working as expected

- [ ] Build all track pieces normally and individually
- [ ] Build connected track
- [ ] Build up/down track
- [ ] Track doesn't build off map
- [ ] Doesn't build into industries
- [ ] Doesn't build into stations/airports
- [ ] Doesn't build into water
- [ ] Builds bridge over water (at correct height)
- [ ] Builds into tunnel
- [ ] Doesn't build into terrain
- [ ] Builds level crossing
- [ ] Doesn't build on parallel road
- [ ] Builds selected bridge types 
- [ ] Clears trees if in way
- [ ] Cost is same as master
- [ ] Track is associated with the correct company
- [ ] Waypoint routing works
- [ ] AI can place track as per master/vanilla
- [ ] All of this also works for tram track
- [ ] All of this also works for roads

Cute little testing playground image
image

**Comment by LeftofZen** [comment]
Proof the AI can also create track still:
image

**Review by LeftofZen — APPROVED** [review]
Couldn't see any issues in playtesting

**Review by AaronVanGeffen — APPROVED** [review]
No issues found in playtesting. I'm trust there are no divergences on the replay/AI front?

**Comment by duncanspumpkin** [comment]
> No issues found in playtesting. I'm trust there are no divergences on the replay/AI front?

all good.

**Pull request merged**